test/unit/auth/token-verifier.spec.ts

```diff
@@ -513,4 +526,305 @@ describe('FirebaseTokenVerifier', () => {
+      stubs.push(verifierStub);
+
+      const mockAuthBlockingToken = mocks.generateAuthBlockingToken();
+
+      return authBlockingTokenVerifier._verifyAuthBlockingToken(mockAuthBlockingToken, false, undefined)
+        .should.eventually.be.rejectedWith('Error fetching public keys.');
+    });
+
+    it('should be rejected given a custom token with error using article "an" before JWT short name', () => {
+      return tokenGenerator.createCustomToken(mocks.uid)
+        .then((customToken) => {
+          return authBlockingTokenVerifier._verifyAuthBlockingToken(customToken, false, undefined)
+            .should.eventually.be.rejectedWith(
+              '_verifyAuthBlockingToken() expects an Auth Blocking token, but was given a custom token');
+        });
+    });
+
+    it('should be rejected given a custom token with error using article "a" before JWT short name', () => {
+      const tokenVerifierSessionCookie = new verifier.FirebaseTokenVerifier(
+        'https://www.googleapis.com/identitytoolkit/v3/relyingparty/publicKeys',
+        'https://session.firebase.google.com/',
+        verifier.SESSION_COOKIE_INFO,
+        app,
+      );
+      return tokenGenerator.createCustomToken(mocks.uid)
+        .then((customToken) => {
+          return tokenVerifierSessionCookie._verifyAuthBlockingToken(customToken, false, undefined)
+            .should.eventually.be.rejectedWith(
+              'verifySessionCookie() expects a session cookie, but was given a custom token');
+        });
+    });
+
+    it('should be rejected given a legacy custom token with error using article "an" before JWT short name', () => {
+      const legacyTokenGenerator = new LegacyFirebaseTokenGenerator('foo');
+      const legacyCustomToken = legacyTokenGenerator.createToken({
+        uid: mocks.uid,
+      });
+
+      return authBlockingTokenVerifier._verifyAuthBlockingToken(legacyCustomToken, false, undefined)
+        .should.eventually.be.rejectedWith(
+          '_verifyAuthBlockingToken() expects an Auth Blocking token, but was given a legacy custom token');
+    });
+
+    it('should be rejected given a legacy custom token with error using article "a" before JWT short name', () => {
+      const tokenVerifierSessionCookie = new verifier.FirebaseTokenVerifier(
+        'https://www.googleapis.com/identitytoolkit/v3/relyingparty/publicKeys',
+        'https://session.firebase.google.com/',
+        verifier.SESSION_COOKIE_INFO,
+        app,
+      );
+      const legacyTokenGenerator = new LegacyFirebaseTokenGenerator('foo');
+      const legacyCustomToken = legacyTokenGenerator.createToken({
+        uid: mocks.uid,
+      });
+
+      return tokenVerifierSessionCookie._verifyAuthBlockingToken(legacyCustomToken, false, undefined)
+        .should.eventually.be.rejectedWith(
+          'verifySessionCookie() expects a session cookie, but was given a legacy custom token');
```

Comment:
Same here... why does `_verifyAuthBlockingToken()` expect a session cookie here?

Reply:
Same - I copied this test from verifyIdToken, removed it in the latest push